### ISSUE TITLE
Private key naming convention has no effect on JavaScript API

### ISSUE THREAD

**Issue by quietopus** [dotenvx/dotenvx#316]
I provided a standard private key like `DOTENV_PRIVATE_KEY_SOMETHING` and loaded the config file from javascript, as follows:

```JavaScript
import {config} from '@dotenvx/dotenvx'
config()
```

(I prefer the command line usage rather than the `config()` function, but I am working in an environment where I cannot control the exact command used to execute my code.)

I expected `.env.something` to be loaded automatically, per [the documentation](https://dotenvx.com/docs/quickstart):

> dotenvx run is smart enough to introspect from either the DOTENV_PRIVATE_KEY_PRODUCTION environment variable or from the .env.keys file.

But this did not happen. After some debug tracing, I found what appears to be the issue.

In summary, the naming convention logic is skipped whenever the `path` property is absent from the config options. The one workaround/exception I found was when (and only when) the `path` is present but contains an empty array, as follows:

```JavaScript
import {config} from '@dotenvx/dotenvx'
config({path:[]})
```

Then it works as expected.

Here is a detailed breakdown of the issue.

1. At [this line](https://github.com/dotenvx/dotenvx/blob/main/src/lib/main.js#L44), `optionPaths` is populated with the output from `dotenvOptionPaths`.
2. At [this line](https://github.com/dotenvx/dotenvx/blob/main/src/lib/helpers/dotenvOptionPaths.js#L4), `dotenvOptionPaths` initializes its output to `['.env']` and, [if there is no `path` value in the options](https://github.com/dotenvx/dotenvx/blob/main/src/lib/helpers/dotenvOptionPaths.js#L6), [returns the initial value](https://github.com/dotenvx/dotenvx/blob/main/src/lib/helpers/dotenvOptionPaths.js#L18).
3. This means that `optionPaths` will always contain `['.env']` when `path` is not specified in the config.
4. Consequently, under those same conditions, [`envs` will always contain `[{ type: 'envFile', value: '.env' }]`](https://github.com/dotenvx/dotenvx/blob/main/src/lib/main.js#L56).
5. [At this line](https://github.com/dotenvx/dotenvx/blob/main/src/lib/main.js#L61), `envs` is passed into the `Run` constructor, which, in turn, [hands it to `_determineEnvs()`](https://github.com/dotenvx/dotenvx/blob/main/src/lib/services/run.js#L22).
6. But `_determineEnvs()` [skips `_determineEnvsFromDotenvPrivateKey()`](https://github.com/dotenvx/dotenvx/blob/main/src/lib/services/run.js#L199-L203) when `envs` is (as we determined it would be) non-empty.

**Comment by motdotla** [comment]
strange. will take a look. you ran this in JavaScript frontend or Node.js?

**Comment by quietopus** [comment]
Node.js

**Comment by motdotla** [comment]
thank you for catching this. fix happening here: https://github.com/dotenvx/dotenvx/pull/352

**Comment by motdotla** [comment]
released as `1.10.2`. let me know if you still run into any issues, but i think you should be good now.
